### PR TITLE
TLDR-688 deleted noqa

### PR DESCRIPTION
**Pull request by alexander1999-hub**
Удалил noqa везде, где смог: 
- в случае недобавленных импортов; 
- в тех местах, где удаление не приводило к возникновению жалоб флэйка; 
- в случаях, где смог понять по коду, на что заменить Any в аннотации функций;
- в эксэспшнах, заменив `except:` на `except Exception`. Решил, что добавлять при этом логи не стоит, т.к. раньше их не было. Если надо добавить, могу это сделать;
- в случаях, где смог правильно перенести часть выражения на следующую строку

Решил не удалять noqa в случаях, где:
- по коду непонятно, на что заменить Any
- в бенчмарках используется `print()`
- в случае длинного названия файла, которое нельзя нормально перенести на следующую строку
- название аттрибута совпадает со встроенным в питон

Не смог пофиксить circular imports, из-за которых возникали ошибки в следующих файлах:
`./dedoc/readers/pdf_reader/pdf_image_reader/ocr/ocr_cell_extractor.py`
`./dedoc/readers/docx_reader/data_structures/run.py`

### CODE REVIEW

File: dedoc/utils/train_dataset_utils.py [hunk 1]
```diff
@@ -30,7 +32,7 @@ def _get_images_path(config: dict, document_name: str) -> str:
     return images_path
 
 
-def save_line_with_meta(lines: List["LineWithMeta"], original_document: str, *, config: dict) -> None:  # noqa
+def save_line_with_meta(lines: List["LineWithMeta"], original_document: str, *, config: dict) -> None:
```
Comment (dronperminov) [inline dedoc/utils/train_dataset_utils.py:35]: кавычки нужно убрать у `LineWithMeta` (посмотри, пожалуйста, в аналогичных местах подобное)

File: scripts/benchmark_table/metric.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 
 class TableTree(Tree):
-    def __init__(self, tag: str, colspan=None, rowspan=None, content=None, visible=None, *children):  # noqa
+    def __init__(self, tag: str, colspan: int = None, rowspan: int = None, content: Iterable[str] = None, visible: bool = None, *children: "TableTree") -> None:
```
Comment (dronperminov) [inline scripts/benchmark_table/metric.py:24]: `int` не может быть `None`

File: scripts/benchmark_table/metric.py [hunk 3]
```diff
@@ -44,13 +44,13 @@ def bracket(self) -> str:
 
 class CustomConfig(Config):
     @staticmethod
-    def maximum(*sequences):  # noqa
+    def maximum(*sequences: str) -> int:
```
Comment (dronperminov) [inline scripts/benchmark_table/metric.py:47]: sequences - не строка, а какая-то итерируемая коллекция (`Iterable`)

Comment (alexander1999-hub) [reply scripts/benchmark_table/metric.py:47]: Исправил

File: scripts/benchmark_table/metric.py [hunk 4]
```diff
@@ -44,13 +44,13 @@ def bracket(self) -> str:
...
         """
         Get maximum possible value
         """
         return max(map(len, sequences))
 
-    def normalized_distance(self, *sequences) -> float:  # noqa
+    def normalized_distance(self, *sequences: str) -> float:
```
[the remark this answers is not in the document]
Comment (alexander1999-hub) [reply scripts/benchmark_table/metric.py:53]: Исправил

File: scripts/train/trainers/base_sklearn_line_classifier.py [hunk 5]
```diff
@@ -28,7 +28,7 @@ class BaseClassifier(XGBClassifier):
     Base class for a classifier.
     See documentation of `XGBClassifier <https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.XGBClassifier>`_ to get more details.
     """
-    def __init__(self, **kwargs: Any) -> None:  # noqa
+    def __init__(self, **kwargs: Dict) -> None:
```
[the remark this answers is not in the document]
Comment (alexander1999-hub) [reply scripts/train/trainers/base_sklearn_line_classifier.py:31]: Исправил

File: scripts/benchmark_table/metric.py (hunk 2 above)
Comment (alexander1999-hub) [reply scripts/benchmark_table/metric.py:24]: Тогда стоит написать `Union[int, None]` вместо просто `int`?

File: dedoc/utils/train_dataset_utils.py (hunk 1 above)
Comment (dronperminov) [reply dedoc/utils/train_dataset_utils.py:35]: да

File: scripts/benchmark_table/metric.py (hunk 2 above)
Comment (dronperminov) [reply scripts/benchmark_table/metric.py:24]: нет, `Optional[int]`

Comment (alexander1999-hub) [reply scripts/benchmark_table/metric.py:24]: Ок

File: dedoc/utils/train_dataset_utils.py (hunk 1 above)
Comment (alexander1999-hub) [reply dedoc/utils/train_dataset_utils.py:35]: Ок

File: dedoc/readers/docx_reader/data_structures/run.py [hunk 6]
```diff
@@ -8,7 +8,7 @@
 
 class Run(BaseProperties):
 
-    def __init__(self, properties: Optional[BaseProperties], styles_extractor: "StylesExtractor") -> None:  # noqa
+    def __init__(self, properties: Optional["BaseProperties"], styles_extractor: "StylesExtractor") -> None:  # noqa
```
Comment (dronperminov) [inline dedoc/readers/docx_reader/data_structures/run.py:11]: По идее тут можно было убрать и вторые кавычки и не добавлять добавленные

File: dedoc/readers/pdf_reader/pdf_image_reader/ocr/ocr_cell_extractor.py [hunk 7]
```diff
@@ -62,7 +62,8 @@ def get_cells_text(self, page_image: np.ndarray, tree_nodes: List["TableTree"],
 
         return self.__create_lines_with_meta(tree_nodes, originalbox_to_fastocrbox, page_image)
 
-    def __handle_one_batch(self, src_image: np.ndarray, tree_table_nodes: List["TableTree"], num_batch: int, language: str = "rus") -> Tuple[OcrPage, List[BBox]]: # noqa
+    def __handle_one_batch(self, src_image: np.ndarray, tree_table_nodes: List["TableTree"], num_batch: int, language: str = "rus") \
```
Comment (dronperminov) [inline dedoc/readers/pdf_reader/pdf_image_reader/ocr/ocr_cell_extractor.py:65]: Здесь не получится TableTree без кавычек написать?

File: dedoc/readers/docx_reader/data_structures/run.py (hunk 6 above)
Comment (alexander1999-hub) [reply dedoc/readers/docx_reader/data_structures/run.py:11]: Да, точно. Убрал

File: dedoc/readers/pdf_reader/pdf_image_reader/ocr/ocr_cell_extractor.py (hunk 7 above)
Comment (alexander1999-hub) [reply dedoc/readers/pdf_reader/pdf_image_reader/ocr/ocr_cell_extractor.py:65]: Нет, возникает цикличный импорт и дедок не собирается

**Review by dronperminov — APPROVED** [review]
Очень крутой и классный пулл реквест! Умничка, @alexander1999-hub!
